src/Provider/Keycloak.php

```diff
@@ -68,11 +83,67 @@ public function __construct(array $options = [], array $collaborators = [])
     }
 
     /**
+     * We need to cache the access token locally allowing for later optional post-processing
+     * by `checkForKeycloakRoles()`
+     *
+     * @param mixed $grant
+     * @param array $options
+     * @return AccessToken
+     */
+    public function getAccessToken($grant, array $options = [])
```

Comment:
I would prefer that we not cache the access token in the client. I will provide comments on the methods that need this token to better understand how to handle the functionality.